main.tf

```diff
@@ -3,61 +3,59 @@ resource "aws_cognito_user_pool" "this" {
 
   sms_authentication_message = var.auth_sms_message
   email_verification_message = var.html_email_message == "" ? var.auth_sms_message : var.html_email_message
-  email_verification_subject = "Código de verificación"
-  mfa_configuration          = "OPTIONAL"
+  email_verification_subject = var.email_verification_subject
 
-  admin_create_user_config {
-    allow_admin_create_user_only = false
-  }
-
-  software_token_mfa_configuration {
-    enabled = true
-  }
+  auto_verified_attributes = var.auto_verified_attributes
 
-  auto_verified_attributes = ["email"]
-
-  device_configuration {
-    challenge_required_on_new_device      = true
-    device_only_remembered_on_user_prompt = false
+  dynamic "software_token_mfa_configuration" {
+    for_each = var.enable_software_token_mfa ? [1] : []
+    content {
+      enabled = true
+    }
   }
 
   email_configuration {
-    email_sending_account = "DEVELOPER"
-    source_arn            = var.ses_configuration_arn
-    from_email_address    = var.ses_from_email
+    email_sending_account  = var.email_sending_account
+    reply_to_email_address = var.reply_to_email_address
+    source_arn             = var.ses_configuration_arn
+    from_email_address     = var.ses_from_email
+  }
+
+  dynamic "sms_configuration" {
+    for_each = var.enable_sms_sns ? [1] : []
+    content {
+      external_id    = var.sms_external_id
+      sns_caller_arn = var.sms_sns_caller_arn
+      sns_region     = var.sms_sns_region
+    }
   }
 
   account_recovery_setting {
     recovery_mechanism {
-      name     = "verified_email"
-      priority = 2
+      name     = var.email_recovery_name
+      priority = var.email_recovery_priority
     }
-
     recovery_mechanism {
-      name     = "verified_phone_number"
-      priority = 1
+      name     = var.phone_number_recovery_name
+      priority = var.phone_number_recovery_priority
     }
   }
 
-  username_attributes = ["email"]
+  username_attributes = var.username_attributes
   username_configuration {
-    case_sensitive = false
+    case_sensitive = var.username_case_sensitive
   }
 
   password_policy {
-    minimum_length    = 6
-    require_lowercase = false
-    require_uppercase = false
-    require_symbols   = false
-    require_numbers   = false
+    minimum_length = var.password_minimum_length
   }
 
   schema {
     name                     = "email"
     attribute_data_type      = "String"
     developer_only_attribute = false
     mutable                  = true
-    required                 = true
+    required                 = false
```

Comment:
Atributo que deberia tener un valor por defecto en una variable

Reply:
Added the default_attribute_required variable in variables.tf with a default value (false), and used it in main.tf to set the required attribute in the Cognito schemas.